### PR TITLE
[core,arm] Various fixes and adjustements for #9363

### PR DESCRIPTION
**Pull request by hardening**
Treat the case where the publicIpAddress is there but empty. 
Don't try RDSTLS if the password has not been been provided.
When RDSTLS is what we will do, disengage all the other security kinds.

### CODE REVIEW

File: libfreerdp/core/gateway/arm.c [hunk 1]
```diff
@@ -699,54 +704,59 @@ static BOOL arm_fill_gateway_parameters(rdpArm* arm, const char* message, size_t
...
-		if (!redirectedAuthGuid)
-			break;
-
-		size_t wGUID_len = 0;
-		wGUID = ConvertUtf8ToWCharAlloc(redirectedAuthGuid, &wGUID_len);
-		if (!wGUID)
+		/* note: we retrieve some more fields for RDSTLS only if we have a password provided by the
+		 * user, otherwise these are useless: we will not be able to do RDSTLS
+		 */
+		do
```
Comment (akallabeth) [inline libfreerdp/core/gateway/arm.c:712]: maybe split this out to a `static` function? the `do {} while(FALSE)` grouping is probably easier to implement that way

**Comment by akallabeth** [comment]
@freerdp-bot test
@hardening is this ready now?

**Comment by hardening** [comment]
PR has been updated and is ready

**Review by akallabeth — APPROVED** [review]
LGTM

**Pull request merged**